pdg/src/info.rs

```diff
@@ -91,15 +91,15 @@ fn get_last_desc(g: &mut Graph) -> HashMap<NodeId, NodeId> {
 /// Finds the inverse of a [`Graph`], each [`Node`] mapping to a list of its children.
 fn collect_children(g: &Graph) -> HashMap<NodeId, Vec<NodeId>> {
     let mut m = HashMap::new();
+    for par in g.nodes.indices() {
+        let _ = m.try_insert(par, Vec::new());
+    }
     for (par, chi) in g
         .nodes
         .iter_enumerated()
         .filter_map(|(idx, node)| Some((node.source?, idx)))
     {
-        m.entry(par).or_insert_with(Vec::new).push(chi)
-    }
-    for par in g.nodes.indices() {
-        let _ = m.try_insert(par, Vec::new());
+        m.get_mut(&par).unwrap().push(chi);
```

Comment:
i think you can `map` here instead of unwrap

Reply:
we're guaranteed the unwrap will work, so isn't unwrap better here because it's simpler?

Reply:
i think `map` is better because it decouples this line of code from the knowledge of what the code above it does

Reply:
How would `.map` be used here?

Reply:
would `m.get_mut(&par).map(|ref mut v| v.push(chi); v)` or not work? but on second thought maybe `unwrap` is fine because the map option relies on side affects

Reply:
Yeah, `map` is generally used for mapping to a value, not running a side effect.  The equivalent to the `map` but more idiomatic would be an `if let Some`.  But we also want to assert here that it's `Some`, right?  At the very least, `unwrap` serves as documentation that it will always be `Some`.

Reply:
that sounds reasonable